endpoints/internal-check-file-acl.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+namespace Automattic\VIP\Files\Acl;
+
+require_once __DIR__ . '/../files/acl/pre-wp-utils.php';
+
+$file_request_uri = $_SERVER['HTTP_X_ORIGINAL_URI'] ?? null;
+
+if ( ! $file_request_uri ) {
+	trigger_error( 'VIP Files ACL failed due to empty URI', E_USER_WARNING );
+
+	http_response_code( 500 );
+
+	exit;
+}
+
+$file_path = parse_url( $file_request_uri, PHP_URL_PATH );
+
+$is_valid_path = Pre_WP_Utils\validate_path( $file_path );
+if ( ! $is_valid_path ) {
+	http_response_code( 500 );
```

Comment:
Should this be an HTTP `400` instead? Here, this state represents a bad request from the client. Above, where we check if there's a URI at all makes sense to be a 500, b/c that's an error on our end.

Reply:
Agreed that 400 is more "correct". I've asked @alexcriss to confirm if that's possible as we're pretty limited in what response codes we can use.